kyuubi-server/web-ui/src/views/engine/index.vue

```diff
@@ -0,0 +1,166 @@
+          :placeholder="$t('share_level')"
+          clearable
+          style="width: 210px"
+          @change="getList">
+          <el-option
+            v-for="item in getShareLevel()"
+            :key="item"
+            :label="item"
+            :value="item" />
+        </el-select>
+        <el-input
+          v-model="searchParam['hive.server2.proxy.user']"
+          :placeholder="$t('user')"
+          style="width: 210px"
+          @keyup.enter="getList" />
+        <el-button type="primary" icon="Search" @click="getList" />
+      </el-space>
+    </header>
+  </el-card>
+  <el-card class="table-container">
+    <el-table v-loading="loading" :data="tableData" style="width: 100%">
+      <el-table-column
+        prop="instance"
+        :label="$t('engine_address')"
+        min-width="20%" />
+      <el-table-column :label="$t('engine_id')" min-width="20%">
+        <template #default="scope">
+          <span>{{
+            scope.row.attributes && scope.row.attributes['kyuubi.engine.id']
+              ? scope.row.attributes['kyuubi.engine.id']
+              : '-'
+          }}</span>
+        </template>
+      </el-table-column>
+      <el-table-column
+        prop="engineType"
+        :label="$t('engine_type')"
+        min-width="20%" />
+      <el-table-column
+        prop="sharelevel"
+        :label="$t('share_level')"
+        min-width="20%" />
+
+      <el-table-column prop="user" :label="$t('user')" min-width="15%" />
+      <el-table-column prop="version" :label="$t('version')" min-width="15%" />
+      <el-table-column fixed="right" :label="$t('operation.text')" width="120">
+        <template #default="scope">
+          <el-space wrap>
+            <el-popconfirm
+              :title="$t('operation.delete_confirm')"
+              @confirm="handleDeleteEngine(scope.row)">
+              <template #reference>
+                <span>
+                  <el-tooltip
+                    effect="dark"
+                    :content="$t('operation.delete')"
+                    placement="top">
+                    <template #default>
+                      <el-button type="danger" icon="Delete" circle />
+                    </template>
+                  </el-tooltip>
+                </span>
+              </template>
+            </el-popconfirm>
+          </el-space>
+        </template>
+      </el-table-column>
+    </el-table>
+  </el-card>
+</template>
+
+<script lang="ts" setup>
+  import { reactive } from 'vue'
+  import { getAllEngines, deleteEngine } from '@/api/engine'
+  import { IEngineSearch } from '@/api/engine/types'
+  import { useTable } from '@/views/common/use-table'
+  import { ElMessage } from 'element-plus'
+  import { useI18n } from 'vue-i18n'
+  import { getEngineType, getShareLevel } from '@/utils/engine'
+
+  const { t } = useI18n()
+  const { tableData, loading, getList: _getList } = useTable()
+  // default search params
+  const searchParam: IEngineSearch = reactive({
```

Comment:
let's display all engines in default

Reply:
Currently, list engines api doesn't support list all engines.

Even if no search parameters are passed, the default configuration in the `Kyuubi Configuration` is used to set up the request.